common/parallel/TreeCrawler.go

```diff
@@ -80,8 +97,18 @@ func (c *crawler) start(ctx context.Context, root Directory) {
 	}
 	go heartbeat()
 
-	c.unstartedDirs = append(c.unstartedDirs, root)
+	if !c.isSource && c.tqueue != nil {
```

Comment:
Again, this should be

if !c.isSourceTraverser {
//
// Target traverser should get directories-to-process from tqueue, where source traverser enqueues them.
//
if c.tqueue == nil {
panic("tqueue is nil in target traverser!")
}
}

PS: We also have to think about source/target traversers which we have not yet updated with the new sync algo.